speculation-rules/prerender/resources/utils.js

```diff
@@ -145,22 +145,44 @@ function createFrame(url) {
 }
 
 class PrerenderChannel extends EventTarget {
-  broadcastChannel = null;
+  #count = 0;
+  #messages = [];
+  #active = false;
+  #baseUrl;
+  #name;
 
   constructor(uid, name) {
     super();
-    this.broadcastChannel = new BroadcastChannel(`${uid}-${name}`);
-    this.broadcastChannel.addEventListener('message', e => {
-      this.dispatchEvent(new CustomEvent('message', {detail: e.data}));
-    });
+    this.#name = name;
+    this.#baseUrl = `/speculation-rules/prerender/resources/channel.py?uid=${uid}&name=${name}`;
+  }
+
+  async #poll() {
+    if (!this.#active)
+      return;
+
+    const response = await fetch(this.#baseUrl);
+    const messages = await response.json();
```

Comment:
I'm somewhat confused. It looks like this is intended to be able to both send and receive on the same channel, but I don't see any logic that prevents the channel from seeing its own outgoing messages echoed back to it, which BroadcastChannel does (and callers would reasonably expect). Am I missing it?

Reply:
It's currently actually a very crude channel where the sender and the receiver can't be the same instance, and doesn't try to be general purpose. Perhaps that makes its API surface confusing. But I can iterate on it and hopefully make it work with `dispatcher`.

Reply:
Made a new more sturdy version

Reply:
OK, I refactored the whole thing to use the `dispatcher` infra.